cmake/windows-setup.cmake

```diff
@@ -20,7 +20,9 @@ if(FLB_WINDOWS_DEFAULTS)
   set(FLB_AWS                   Yes)
   set(FLB_HTTP_SERVER           Yes)
   set(FLB_METRICS               Yes)
-  set(FLB_CONFIG_YAML           No)
+  if (NOT FLB_LIBYAML_DIR)
```

Comment:
I'd be tempted to remove it entirely, let's move away from hidden disables and leave the global setting if we can.

Reply:
Ideally, I'd like to do so. However, AppVeyor build does not handle libyaml on building and libyaml requirements forcibly makes hard to build on Windows for now...

Reply:
Another reason to kill AppVeyor to be honest.
The actual AppVeyor set up is slightly hidden away: https://github.com/fluent/fluent-bit-ci/blob/main/scripts/run-unit-tests.ps1